docs/rag/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,438 @@
+
+### 1.1 User Experience Overview
+
+The commands are tailored to support diverse user experiences, all enabling the use of RAG functionality to enrich chat sessions.
+
+### 1.2 Model Training Path
+
+This flow is designed for users who aim to train their own models and leverage the source documents that support knowledge submissions to enhance the chat context:
+![model-training](./images/rag-model-training.png)
+
+**Note**: documents are processed using `instructlab-sdg` package and are defined using the docling v1 schema.
+
+### 1.3 Taxonomy Path (no Training)
+
+This flow is for users who have defined taxonomy knowledge but prefer not to train their own models. Instead, they aim to generate RAG artifacts from source documents to enhance the chat context:
+![taxonomy-no-training](./images/rag-taxonomy-no-training.png)
+
+**Note**: documents are processed using `docling.DocumentConverter` and are defined using the docling v2 schema.
+
+### 1.4 Plug-and-Play RAG Path
+
+This flow is designed for users who want to enhance their chat experience with pre-trained models by simply integrating the RAG functionality:
+![plug-and-play](./images/rag-plug-and-play.png)
+
+**Note**: documents are processed using `docling.DocumentConverter` and are defined using the docling v2 schema.
+
+## 2. Proposed Pipelines
+
+### 2.1 Working Assumption
+
+This proposal aims to serve as a reference design to develop a Proof of Concept for RAG workflows, while
+also laying the foundation for future implementations of state-of-the-art RAG artifacts tailored to specific use
+cases.
+
+#### Command Options
+
+To maintain compatibility and simplicity, no new configurations will be introduced for new commands. Instead,
+the settings will be defined using the following hierarchy (options higher in the list overriding those below):
+
+* CLI flags (e.g., `--FLAG`).
+* Environment variables following a consistent naming convention, such as `ILAB_<UPPERCASE_FLAG_NAME>`.
+* Default values, for all the applicable use cases.
+
+For example, the `vectordb-uri` argument can be implemented using the `click` module like this:
+
+```py
+@click.option(
+    "--document-store-uri",
+    default='rag-output.db',
+    envvar="ILAB_DOCUMENT_STORE_URI",
+)
+```
+
+#### Local embedding models
+
+The embedding model used to generate text embeddings must be downloaded locally before executing the pipeline.
+
+For example, this command can be used to download the `ibm-granite/granite-embedding-125m-english` model to the local models cache:
+
+```bash
+ilab model download -rp ibm-granite/granite-embedding-125m-english
+```
+
+If the configured embedding model has not been cached, the command execution will terminate with an error. This requirement applies
+consistently to all new and updated commands.
+
+### 2.2 Document Processing Pipeline
+
+The proposal is to add a `convert` sub-command to the `rag` command group.
```

Comment:
I will note there is an existing `ilab model convert` command - I think we can still use `convert` here, but if another term works for this purpose that may be simpler for users